aten/src/ATen/cpu/vec/functional_base.h

```diff
@@ -83,7 +83,8 @@ inline scalar_t vec_reduce_all(const Op& vec_fun, const Vectorized<scalar_t>& ac
   return VecReduceAllSIMD<scalar_t, Op>::apply(vec_fun, acc_vec);
 }
 
-template <typename scalar_t, typename Op>
+template <typename scalar_t, typename Op,
+          typename std::enable_if<!is_reduced_floating_point<scalar_t>::value, int>::type = 0>
```

Comment:
In typical C++17 fashion, can you please add `is_reduced_floating_point_v` constexpr to avoid typing ::value. 
Also, use `enable_if_t` which is available since C++14
```suggestion
          typename std::enable_if_t<!is_reduced_floating_point<scalar_t>::value, int> = 0>
```

Reply:
Thanks for the hint! Updated!